altair/utils/core.py

```diff
@@ -767,7 +769,132 @@ def display_traceback(in_ipython: bool = True):
         traceback.print_exception(*exc_info)
 
 
-def infer_encoding_types(args: Sequence, kwargs: MutableMapping, channels: ModuleType):
+_ChannelType = Literal["field", "datum", "value"]
+_CHANNEL_CACHE: "_ChannelCache"
+"""Singleton `_ChannelCache` instance.
+
+Initialized on first use.
+"""
+
+
+class _ChannelCache:
+    channel_to_name: Dict[Type[SchemaBase], str]
+    name_to_channel: Dict[str, Dict[_ChannelType, Type[SchemaBase]]]
+
+    @classmethod
+    def from_channels(cls, channels: ModuleType, /) -> "_ChannelCache":
```

Comment:
As stated in the comments, I've kept this for an existing test https://github.com/dangotbanned/altair/blob/e4ab7052e9a1e62ff1fd80379864489c69a1e020/tests/utils/test_core.py#L263

I would prefer it if this was removed and `test_infer_encoding_types` was re-written using actual objects from `channels`.